File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int numSwaps = 0;
        for (int i = 0; i < nums.length; i++) {
          for (int j = 0; j < nums.length - 1; j++) {
            if (nums[j] > nums[j + 1]) {
              // swap them
              numSwaps += 1;
              int tmp = nums[j];
              nums[j] = nums[j + 1];
              nums[j+1] = tmp;
            }
          }
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # INSERTION SORT
        # TIME: O(N^2) Worst Case & Average Case. O(N) Best Case
        # SPACE: O(1) because in place.

        # start with the second index
        # compare current item with its preceding elements, stopping when it reaches an element it is smaller than
        # once we've found our insertion point, insert
        for i in range(1, len(nums)):
          tmp = nums[i]
          j = i - 1
          # break when nums[j] is smaller than tmp
          while j >= 0 and nums[j] > tmp:
            # shift elements to the right
            nums[j + 1] = nums[j]
            j -= 1
          # we have a hole to right of our current j. this is where tmp goes
          nums[j + 1] = tmp
        

            



        